### PR TITLE
Split resume function for Audiobooks

### PR DESCRIPTION
**Pull request by Artiume**
Web PR: https://github.com/jellyfin/jellyfin-web/pull/2286
test
**Changes**

Split the resume logic for audiobooks from the rest of the media
**Issues**

https://github.com/jellyfin/jellyfin/issues/4897

### CODE REVIEW

File: Emby.Server.Implementations/Library/UserDataManager.cs [hunk 1]
```diff
@@ -245,6 +246,24 @@ public bool UpdatePlayState(BaseItem item, UserItemData data, long? reportedPosi
                     }
                 }
             }
+            else if (positionTicks > 0 && hasRuntime && (item is AudioBook))
+            {
+                var minIn = TimeSpan.FromTicks(positionTicks).TotalSeconds / 60;
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/UserDataManager.cs:251]:
```suggestion
                var minIn = TimeSpan.FromTicks(positionTicks).TotalMinutes;
```

File: Emby.Server.Implementations/Library/UserDataManager.cs [hunk 2]
```diff
@@ -245,6 +246,24 @@ public bool UpdatePlayState(BaseItem item, UserItemData data, long? reportedPosi
...
+                // 10,000 ticks per millisecond * 60,000 milliseconds per minute = 600,000,000 ticks per minute
+                var minOut = (runtimeTicks - positionTicks) / 600000000;
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/UserDataManager.cs:253]:
```suggestion
                var minOut = TimeSpan.FromTicks(runtimeTicks - positionTicks).TotalMinutes;
```

File: MediaBrowser.Model/Configuration/ServerConfiguration.cs [hunk 3]
```diff
@@ -298,6 +298,18 @@ public ServerConfiguration()
         /// <value>The min resume duration seconds.</value>
         public int MinResumeDurationSeconds { get; set; } = 300;
 
+        /// <summary>
+        /// Gets or sets the minimum minutes of a book that must be played in order for playstate to be updated.
```
Comment (Bond-009) [inline MediaBrowser.Model/Configuration/ServerConfiguration.cs:302]: What's wrong with the default that all other media uses?

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.Model/Configuration/ServerConfiguration.cs (hunk 3 above)
Comment (Artiume) [reply MediaBrowser.Model/Configuration/ServerConfiguration.cs:302]: A 13 hour audiobook will be marked finished when theres about an hour left to listen to. It gets worse as the length gets longer. Percentages dont really work well.

File: Emby.Server.Implementations/Library/UserDataManager.cs [hunk 4]
```diff
@@ -245,6 +246,23 @@ public bool UpdatePlayState(BaseItem item, UserItemData data, long? reportedPosi
                     }
                 }
             }
+            else if (positionTicks > 0 && hasRuntime && (item is AudioBook))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/UserDataManager.cs:249]:
```suggestion
            else if (positionTicks > 0 && hasRuntime && item is AudioBook)
```

**Review by Bond-009 — APPROVED** [review]

**Review by BaronGreenback — APPROVED** [review]

**Pull request merged**